File: langgraph/data/equiv/127.py
def check_interval_for_prime_length(range1, range2):
    def prime_check(number):
        if number in (0, 1):
            return False
        if number == 2:
            return True
        for divisor in range(2, number):
            if number % divisor == 0:
                return False
        return True

    start = max(range1[0], range2[0])
    end = min(range1[1], range2[1])
    span = end - start
    if span > 0 and prime_check(span):
        return "YES"
    return "NO"

# method name: intersection -> check_interval_for_prime_length
# parameter name: interval1 -> range1, interval2 -> range2
# inner function name: is_prime -> prime_check
# variable name: l -> start, r -> end, length -> span
# used tuple unpacking for clarity in max and min functions
# kept the logic and structure similar but changed identifiers and some syntax for clarity

File: langgraph/result/qwen/3b-eq2js/127.js
function checkIntervalForPrimeLength(range1, range2) {
    function primeCheck(number) {
        if (number === 0 || number === 1) return false;
        if (number === 2) return true;
        for (let divisor = 2; divisor < number; divisor++) {
            if (number % divisor === 0) return false;
        }
        return true;
    }

    const [start, end] = range1.length > range2.length ? range1 : range2;
    const span = end - start;
    if (span > 0 && primeCheck(span)) {
        return "YES";
    }
    return "NO";
}